File: gestion_compte_heritage/src/entity/Compte.java
package entity;

import java.time.LocalDate;
import java.util.ArrayList;

import services.FormatDate;
import views.translate.fr.MessageTransaction;

public class Compte {
    //Attributs d'instances
    protected int id; //0
    protected String numero;//null
    protected  LocalDate dateCreation=LocalDate.now();//null
    protected double  solde;//0.0
    protected  ArrayList<Transaction> transactions=new ArrayList<>();
    protected TypeCompte type;

  

    public  ArrayList<Transaction> getTransactions() {
        return transactions;
    }


    public TypeCompte getType() {
        return type;
    }
    public  MessageTransaction retrait(Transaction transaction){
        return null;
    }

    public  MessageTransaction depot(Transaction transaction){
        return null;
    }


    //Attributs 
    private static int nbreCompte;//0

    //Constructeur
    public Compte() {
        nbreCompte++;
        id=nbreCompte;
    }

    public static int getNbreCompte() {
        return nbreCompte;
    }

    public static void setNbreCompte(int nbreCompte) {
        Compte.nbreCompte = nbreCompte;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNumero() {
        return numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

    public LocalDate getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(LocalDate dateCreation) {
        this.dateCreation = dateCreation;
    }

    public double getSolde() {
        return solde;
    }

    public void setSolde(double solde) {
        this.solde = solde;
    }

    @Override
    public String toString() {
        return " [id=" + id + ", numero=" + numero + ", dateCreation=" + FormatDate.toFr(dateCreation) + ", solde=" + solde + "]";
    }


 
}

File: php_gestion_compte/src/models/Compte.php
<?php 
class Compte{
       private static int $nbreCompte=0;
       private int $id;
       private string $numero;
       private DateTime $dateCreation;
       private string $solde;

      //PHP Constructeur par defaut ==>  __construct()
      public function __construct()
      {
        self::$nbreCompte++;
        $this->id=self::$nbreCompte;
        $this->numero="NUM_". $this->id;
        $this->dateCreation=new DateTime();

      }

     //Java ==>this.nomAttribut
        //PHP ==>$this->nomAttribut  ==> Obligatoire sur les attributs
        //PHP ==>$this->nomMethode()  ==> Obligatoire sur les attributs
       /**
        * Get the value of id
        */
       public function getId(): int
       {
              return $this->id;
       }

       /**
        * Set the value of id
        */
       public function setId(int $id): void
       {
              $this->id = $id;
       }

       /**
        * Get the value of numero
        */
       public function getNumero(): string
       {
              return $this->numero;
       }

       /**
        * Set the value of numero
        */
       public function setNumero(string $numero): void
       {
              $this->numero = $numero;
       }

       /**
        * Get the value of dateCreation
        */
       public function getDateCreation(): DateTime
       {
              return $this->dateCreation;
       }

       /**
        * Set the value of dateCreation
        */
       public function setDateCreation(DateTime $dateCreation): void
       {
              $this->dateCreation = $dateCreation;

  
       }

       /**
        * Get the value of solde
        */
       public function getSolde(): string
       {
              return $this->solde;
       }

       /**
        * Set the value of solde
        */
       public function setSolde(string $solde):void
       {
              $this->solde = $solde;
       }

       public function __toString()
       {
           return "ID :".$this->id."\n Numero :".$this->numero."\n Numero :".$this->dateCreation->format("d-m-Y")."\n Solde :".$this->solde."\n";
       }
}